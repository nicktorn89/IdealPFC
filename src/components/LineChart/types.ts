export interface LineChartProps {
  data: LineChartData[];
}

export type LineChartData = {
  name: string;
  calories: number;
};
