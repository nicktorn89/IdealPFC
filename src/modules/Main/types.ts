export interface MainState {
  weight: number;
  height: number;
  age: number;
  desiredWeight?: number;

  [key: string]: number | undefined;
}

export interface MainProps {
}

export interface Nutrients {
  protein: number;
  fats: number;
  carbohydrates: number;
}
