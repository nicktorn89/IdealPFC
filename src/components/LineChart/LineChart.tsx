import React from 'react';
import { Section } from 'react-bulma-components/full';
import { LineChartProps } from './types';
import { LineChart as VendorLineChart, Line, Legend, Tooltip, YAxis, XAxis, CartesianGrid } from 'recharts';

const LineChart: React.FC<LineChartProps> = ({ data }) =>
  <Section>
    <VendorLineChart
      width={730}
      height={250}
      data={data}
      margin={{ top: 5, right: 30, left: 20, bottom: 5 }}
    >
      <CartesianGrid strokeDasharray='3 3' />
      <XAxis dataKey='name' tickMargin={20} />
      <YAxis unit='ккал' domain={[0, 3000]} tickMargin={20} />
      <Tooltip />
      <Legend margin={{ top: 50 }} />

      <Line type='monotone' dataKey='calories' stroke='#82ca9d' />
    </VendorLineChart>
  </Section>;

export default LineChart;
