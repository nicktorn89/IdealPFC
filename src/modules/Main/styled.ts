import styled from 'styled-components';
import { media } from 'src/theme/mixin';
