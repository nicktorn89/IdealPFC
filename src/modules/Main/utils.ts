import { Nutrients } from './types';

export const roundToDecimal = (value: number) => Math.round(value * 10) / 10;

export const countNutrients = (height: number, weight: number, hasSportLoad?: boolean): Nutrients => 
  ({
    protein: roundToDecimal(weight * (hasSportLoad ? 2.5 : 1.5)),
    fats: roundToDecimal(weight * (hasSportLoad ? 1.5 : 0.8)),
    carbohydrates: roundToDecimal(weight * (hasSportLoad ? 4 : 2)),
  });
