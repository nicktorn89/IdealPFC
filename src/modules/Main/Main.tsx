import React from 'react';
import { MainState, MainProps } from './types';
import FormComponent from '../../components/Form';
import HeaderComponent from 'src/components/Header';
import ChartPFC from 'src/components/ChartPFC';
import { PieChartData } from 'src/components/ChartPFC/types';
import LineChart from 'src/components/LineChart';
import { LineChartData } from 'src/components/LineChart/types';

const dataMock: PieChartData[] = [
  {
    name: 'Белки',
    value: 200,
    color: '#FFFF4C',
  },
  {
    name: 'Жиры',
    value: 150,
    color: '#FF4C4C',
  },
  {
    name: 'Углеводы',
    value: 650,
    color: '#4C4CFF',
  },
];

const anotherMock: LineChartData[] = [
  {
    name: '1 неделя',
    calories: 2500,
  },
  {
    name: '2 неделя',
    calories: 2300,
  },
  {
    name: '3 неделя',
    calories: 2100,
  },
  {
    name: '4 неделя',
    calories: 1900,
  },
  {
    name: '5 неделя',
    calories: 1700,
  },
];

export class Main extends React.Component<MainProps, MainState> {
  public readonly state = {
    weight: 0,
    height: 0,
    age: 0,
  };

  public handleFieldChange = ({ target }: React.MouseEvent<HTMLInputElement>) => {
    const { name, value } = target as HTMLInputElement;

    if (isNaN(+value)) return;

    this.setState({ [name]: +value });
  }

  public render = () => {
    return (
      <React.Fragment>
        <HeaderComponent />
        <FormComponent
          values={this.state}
          onChangeField={this.handleFieldChange}
        />
        <ChartPFC data={dataMock} />
        <LineChart data={anotherMock} />
      </React.Fragment>
    );
  }
}

export default Main;
