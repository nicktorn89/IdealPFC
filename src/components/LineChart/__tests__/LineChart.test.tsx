import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { default as Chart } from '../LineChart';
import { anotherMock } from 'src/modules/Main/Main';

describe('<LineChart />', () => {
  const LineChart = shallow(<Chart data={anotherMock} />);
  const LineChartToJson = shallowToJson(LineChart);

  it('should be equal to snapshot', () => {
    expect(LineChartToJson).toMatchSnapshot();
  });
});
