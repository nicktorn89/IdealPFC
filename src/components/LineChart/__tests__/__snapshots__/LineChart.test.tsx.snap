// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineChart /> should be equal to snapshot 1`] = `
<c
  className=""
  renderAs="section"
  size={null}
  style={Object {}}
>
  <LineChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "calories": 2500,
          "name": "1 неделя",
        },
        Object {
          "calories": 2300,
          "name": "2 неделя",
        },
        Object {
          "calories": 2100,
          "name": "3 неделя",
        },
        Object {
          "calories": 1900,
          "name": "4 неделя",
        },
        Object {
          "calories": 1700,
          "name": "5 неделя",
        },
      ]
    }
    height={250}
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 20,
        "right": 30,
        "top": 5,
      }
    }
    reverseStackOrder={false}
    stackOffset="none"
    width={730}
  >
    <CartesianGrid
      fill="none"
      horizontal={true}
      horizontalFill={Array []}
      horizontalPoints={Array []}
      stroke="#ccc"
      strokeDasharray="3 3"
      vertical={true}
      verticalFill={Array []}
      verticalPoints={Array []}
    />
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      tickMargin={20}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      domain={
        Array [
          0,
          3000,
        ]
      }
      height={0}
      hide={false}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      tickMargin={20}
      type="number"
      unit="ккал"
      width={60}
      yAxisId={0}
    />
    <Tooltip
      active={false}
      animationDuration={400}
      animationEasing="ease"
      contentStyle={Object {}}
      coordinate={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      cursor={true}
      cursorStyle={Object {}}
      filterNull={true}
      isAnimationActive={true}
      itemStyle={Object {}}
      labelStyle={Object {}}
      offset={10}
      separator=" : "
      useTranslate3d={false}
      viewBox={
        Object {
          "x1": 0,
          "x2": 0,
          "y1": 0,
          "y2": 0,
        }
      }
      wrapperStyle={Object {}}
    />
    <Legend
      align="center"
      iconSize={14}
      layout="horizontal"
      margin={
        Object {
          "top": 50,
        }
      }
      verticalAlign="bottom"
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey="calories"
      dot={true}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="#82ca9d"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
  </LineChart>
</c>
`;
